spark3/src/main/java/org/apache/spark/sql/connector/iceberg/catalog/SupportsMerge.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.spark.sql.connector.iceberg.catalog;
+
+import org.apache.spark.sql.connector.catalog.Table;
+import org.apache.spark.sql.connector.iceberg.write.MergeBuilder;
+import org.apache.spark.sql.connector.write.LogicalWriteInfo;
+
+public interface SupportsMerge extends Table {
```

Comment:
Can we add Javadoc for all of these classes and methods? I think that would be helpful.

Reply:
Yeah, let me spend some time to add docs to the existing classes too. I think we are relatively confident in the implementation now.

Reply:
I added some basic docs. Feel free to correct wording if it is not clear.